x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/image_upload/index.js

```diff
@@ -98,74 +107,70 @@ class ImageUpload extends React.Component {
     onValueChange(srcUrl);
   };
 
-  urlTypeOptions = [
-    { value: 'inline', text: 'Upload Image' },
-    { value: 'src', text: 'Paste Image URL' },
-  ];
-
   render() {
     const { loading, url, urlType } = this.state;
-    const urlTypeInline = urlType === 'inline';
-    const urlTypeSrc = urlType === 'src';
+    const assets = Object.values(this.props.workpad.assets);
+
+    let selectedAsset = {};
+
+    const urlTypeOptions = [{ id: 'file', label: 'Import' }, { id: 'link', label: 'Link' }];
+    if (assets.length) {
+      urlTypeOptions.unshift({ id: 'asset', label: 'Asset' });
+      selectedAsset = assets.find(({ value }) => value === url);
+      console.log({ selectedAsset });
+    }
 
     const selectUrlType = (
-      <EuiSelect
-        compressed
-        options={this.urlTypeOptions}
-        value={urlType}
+      <EuiButtonGroup
+        buttonSize="s"
+        options={urlTypeOptions}
+        idSelected={urlType}
         onChange={this.changeUrlType}
+        isFullWidth
       />
     );
 
-    let uploadImage = null;
-    if (urlTypeInline) {
-      uploadImage = loading ? (
+    const forms = {
+      file: loading ? (
```

Comment:
nit: this component is getting pretty large now, having 3 different "image upload form" components in that folder instead, and using them here, might be a nice change.